lib/jeckle/http.rb

```diff
@@ -72,8 +92,12 @@ def api_mapping
       end
 
       def run_request(endpoint, options = {})
-        Jeckle::Request.run api_mapping[:default_api], endpoint, options
+        Jeckle::Request.run(api_mapping[:default_api], endpoint, options).tap do |request|
+          if logger = api_mapping[:default_api].logger
```

Comment:
@brennovich What do you think about extracting `logger` to a private method?

Also, the way that it is implemented now, the assignment will always happen. Shouldn't it be `||=`?